cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -1478,6 +1477,7 @@ private void deployDummyAppWithImpersonation(NamespaceMeta nsMeta, @Nullable Str
   public void afterTest() throws Exception {
     Authorizer authorizer = getAuthorizer();
 
+    SecurityRequestContext.setUserId(ALICE.getName());
```

Comment:
So that test failures get isolated. Otherwise a single test failing will cascade onto other tests during `afterTest()` since the principal may not be reset to ALICE who has privileges to clean up entities.